### PR TITLE
create pkg-config .pc file when installing (with cmake >= 3.12)

### PR DESCRIPTION
**Pull request by wdouglass**
This makes libmultisense easier to integrate with multiple (cmake and non-cmake) build systems

### CODE REVIEW

**Review by oamidi — COMMENTED** [review]
Hi. 

I rather not comment on anything libmultisense related.
I don't use it and know just enough about it to be dangerous.

Please rely on Matt A and Patrick Smith to comment.

Thanks
-- Omead

**Review by jyakubik — CHANGES_REQUESTED** [review]
Really only the description needs fixed... otherwise looks fine

File: CMakeLists.txt [hunk 1]
```diff
@@ -93,6 +103,25 @@ endif ()
...
+Name: ${PROJECT_NAME}
+Description: ${CMAKE_PROJECT_DESCRIPTION}
+URL: ${CMAKE_PROJECT_HOMEPAGE_URL}
+Version: ${PROJECT_VERSION}
+Cflags: -I$<JOIN:\"\${includedir}\";$<TARGET_PROPERTY:MultiSense,INTERFACE_SYSTEM_INCLUDE_DIRECTORIES>, -I>
+Libs: -L\"\${libdir}\" -lMultiSense
+")
+install(FILES ${CMAKE_BINARY_DIR}/LibMultiSense.pc DESTINATION lib/pkgconfig COMPONENT development)
+endif (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12")
+
```
[the remark this answers is not in the document]
Comment (wdouglass) [reply CMakeLists.txt:124]: I think so -- it's harmless, and some crazy person may use pkgconfig under msys or something

File: CMakeLists.txt [hunk 2]
```diff
@@ -93,6 +103,25 @@ endif ()
 
 add_subdirectory(source)
 
+#
+# Generate pc file, for pkgconfig compatibility
+#
+if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12")
```
[the remark this answers is not in the document]
Comment (jyakubik) [reply CMakeLists.txt:109]: That's fair...

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jyakubik) [reply CMakeLists.txt:124]: i hope we never have to support a mingw build under windows...

**Review by jyakubik — APPROVED** [review]

**Comment by mattalvarado** [comment]
Built and tested locally. This works for me!

**Pull request merged**